### PR TITLE
fix(training): handle unknown GPU metrics

### PR DESCRIPTION
**Pull request by Dayuxiaoshui**
## What

This change improves training compatibility on GPUs whose peak BF16 throughput is not registered in Relax, including the NVIDIA GeForce RTX 5090.

- Forward `masked_softmax_fusion` from parsed training arguments to the Megatron Bridge model provider.
- Omit `perf/device_peak_tflops` and MFU metrics when the detected device peak throughput is non-finite or non-positive.
- Add unit tests for known and unknown GPU peak-throughput cases.

## Why

The single-GPU Qwen3 training recipe disables masked softmax fusion with `--no-masked-softmax-fusion`. In Megatron Bridge mode, only arguments listed in `bridge_keys` are copied to the generated model provider. Because `masked_softmax_fusion` was not in that list, the provider retained its default value and ignored the launch-script override.

Relax also returns `float("inf")` when the current GPU is absent from the peak-throughput table. The performance logger previously included this value as `perf/device_peak_tflops`. The metrics client uses strict JSON serialization, so the non-finite value caused metric batch reporting to fail with:

```text
Out of range float values are not JSON compliant: inf
```

This change keeps training functional on an unknown GPU while avoiding fabricated MFU values.

## How

`masked_softmax_fusion` is added to the existing Megatron Bridge override allowlist. The established provider override loop can therefore apply the parsed CLI value without adding GPU-specific branching.

The performance logger now validates peak throughput with `math.isfinite(peak_tflops)` and `peak_tflops > 0`. It continues to report measured training TFLOPS and token throughput, but only reports device peak TFLOPS and derived MFU values when the denominator is valid.

The implementation does not hard-code an RTX 5090 peak-throughput estimate. Until a verified value is added to the device table, the existing unknown-GPU warning remains and MFU is intentionally omitted.

## Testing

Focused unit tests:

```bash
.venv/bin/python -m pytest -q \
  tests/utils/test_train_metric_utils.py \
  tests/utils/test_flops_counter.py \
  tests/utils/test_metrics_service.py
```

Result:

```text
34 passed, 16 warnings in 3.88s
```

Syntax validation:

```bash
.venv/bin/python -m py_compile \
  relax/backends/megatron/model_provider.py \
  relax/utils/training/train_metric_utils.py \
  tests/utils/test_train_metric_utils.py
```

An end-to-end RTX 5090 smoke run completed one rollout and two optimizer steps. The run reported `train/step` 0 and 1, successfully sent performance metrics, and completed the Ray job without the JSON `inf` error.

- [ ] `pre-commit run --all-files` passes (not run: `pre-commit` is not installed in the current environment)
- [ ] Tests pass (`pytest tests/`) (full suite requires distributed GPU resources and was not run)
- [x] New tests added (if applicable)
- [ ] Documentation updated (if applicable)

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Screenshots / Logs

Megatron Bridge applied the launch-script override:

```text
Override provider.masked_softmax_fusion: True -> False
```

The unknown device warning remains explicit:

```text
Unknown GPU 'NVIDIA GeForce RTX 5090' - MFU will not be reported.
```

The performance payload contains finite measured metrics and omits `perf/device_peak_tflops` and MFU:

```text
'perf/actor_train_tflops': 7.681317767785444,
'perf/actor_train_tok_per_s': 1557.9656880662892,
'perf/step_token_per_s': 218.39528137104486
```

Training and Ray job completion:

```text
step 0: {..., 'train/step': 0}
step 1: {..., 'train/step': 1}
Main func successfully
Job 'raysubmit_1k4RMNj5uptmi8kn' succeeded
```

### CODE REVIEW

**Comment by NINGBENZHE** [comment]
@Dayuxiaoshui Thanks for the fix—this looks good to me!

**Review by NINGBENZHE — APPROVED** [review]
